crates/compiler/src/polyfill/mod.rs

```diff
@@ -53,186 +53,1192 @@
-    /// or returning [`None`] otherwise.
+/// Functionality useful for testing and experimenting with this container.
+#[cfg(test)]
+impl PolyfillMap {
+    /// Instantiates an empty polyfill mapping.
+    ///
+    /// This **should never be used** in the compiler itself, and exists only
+    /// for testing.
     #[must_use]
-    pub fn llvm(&self, polyfill_name: &str) -> Option<&String> {
-        self.mapping.get_by_right(polyfill_name)
+    pub fn empty() -> Self {
+        let mapping = PolyMapping::new();
+        Self { mapping }
     }
 
-    /// Queries for the LLVM opcode (as modified by [`Self::of_opcode`]) that
-    /// corresponds to the provided `polyfill_name`, returning it if it exists
-    /// or returning an error otherwise.
-    ///
-    /// # Errors
-    ///
-    /// - [`Error::MissingPolyfill`] if the requested polyfill does not exist.
-    pub fn llvm_or_err(&self, polyfill_name: &str) -> Result<&String> {
-        self.llvm(polyfill_name).ok_or(Error::missing_polyfill(polyfill_name))
+    /// Adds a specified `operation` and polyfill name pair to the map for
+    /// testing purposes.
+    pub fn insert(&mut self, operation: LLVMOperation, poly_name: &str) {
+        self.mapping.insert(operation, poly_name.to_string());
     }
+}
 
-    /// Queries for the LLVM opcode (as modified by [`Self::of_opcode`]) that
-    /// corresponds to the provided `polyfill_name`, returning it if it exists.
-    ///
-    /// # Panics
-    ///
-    /// If no llvm name exists for the provided `polyfill_name`.
-    #[must_use]
-    pub fn expect_llvm(&self, polyfill_name: &str) -> &String {
-        self.llvm(polyfill_name)
-            .unwrap_or_else(|| panic!("{polyfill_name} llvm definition was not present"))
+/// The definition of the [unary operations](https://llvm.org/docs/LangRef.html#unary-operations).
+impl PolyfillMap {
+    fn all_unary_ops(&mut self) {
+        self.fneg();
     }
 
-    /// Provides more information to assist in resolving the correct polyfill
-    /// based on the types associated with the particular opcode invocation.
-    ///
-    /// Note that this is a purely _syntactic_ transformation, and does not
-    /// account for type aliases and the like. Please ensure that any types are
-    /// fully resolved before calling this.
-    ///
-    /// ```
-    /// use hieratika_compiler::polyfill::PolyfillMap;
-    ///
-    /// let opcode_name = "add";
-    /// let arg_types = vec!["i8", "i64"];
-    ///
-    /// assert_eq!(
-    ///     PolyfillMap::of_opcode(opcode_name, arg_types.as_slice()),
-    ///     "__llvm_add_i8_i64"
-    /// );
-    /// ```
+    fn fneg(&mut self) {
+        for typ in Self::float_types() {
+            self.mk("fneg", &[typ.clone()], typ.clone());
+        }
+    }
+}
+
+/// The definition of the [binary operations](https://llvm.org/docs/LangRef.html#binary-operations).
+impl PolyfillMap {
+    binop!(add, integer_types);
+
```

Comment:
Rustfmt stuck these newlines in here? Might be worth turning that off for this map and getting rid of the blank lines, as IMHO they just hurt readability.

Reply:
There is no good way to disable formatting of these without splitting them out into a separate `impl` block, which would defeat the purpose of keeping the categories together.